Plumber/launcher client sometimes somehow ends up as nil in trigger.go

I'm really confused by this one, but I've seen nil-pointers show up once in a while that seem to be due to what's currently the `c.PlumberClient` being nil in `trigger.runRequested(...)`. This crashes the controller, of course. Trying to debug this now.

Dangit, this showed up again. 

```
{"Branch":"master","Clone":"https://github.com/cloudbees/jx-tenant-service.git","ID":"186003333","Link":"https://github.com/cloudbees/jx-tenant-service","Name":"jx-tenant-service","Namespace":"cloudbees","Webhook":"pull_request","author":"jenkins-x-bot","event-GUID":"e2048b00-849a-11ea-8ac5-7621e2e610d7","job":"pr-build","level":"info","msg":"Creating a new LighthouseJob.","name":"e4267cab-849a-11ea-b03f-cae5cab3f9e1","org":"cloudbees","plugin":"trigger","pr":715,"repo":"jx-tenant-service","time":"2020-04-22T13:12:17Z","type":"presubmit","url":"https://github.com/cloudbees/jx-tenant-service/pull/715.diff"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x1707bff]

goroutine 14995 [running]:
github.com/jenkins-x/lighthouse/pkg/plugins/trigger.runRequested(0x1fb7f00, 0xc000dde100, 0x0, 0x0, 0xc000307000, 0xc000f38850, 0x1f6cce0, 0xc000ca8150, 0xc0012b04d8, 0xc000c52600, ...)
	/workspace/source/pkg/plugins/trigger/trigger.go:258 +0x45f
github.com/jenkins-x/lighthouse/pkg/plugins/trigger.RunAndSkipJobs(0x1fb7f00, 0xc000dde100, 0x0, 0x0, 0xc000307000, 0xc000f38850, 0x1f6cce0, 0xc000ca8150, 0xc0012b04d8, 0xc000c52600, ...)
	/workspace/source/pkg/plugins/trigger/trigger.go:220 +0x2f3
github.com/jenkins-x/lighthouse/pkg/plugins/trigger.buildAll(0x1fb7f00, 0xc000dde100, 0x0, 0x0, 0xc000307000, 0xc000f38850, 0x1f6cce0, 0xc000ca8150, 0xc0012b04d8, 0xc00024cb70, ...)
	/workspace/source/pkg/plugins/trigger/pull-request.go:222 +0x2e3
github.com/jenkins-x/lighthouse/pkg/plugins/trigger.handlePR(0x1fb7f00, 0xc000dde100, 0x0, 0x0, 0xc000307000, 0xc000f38850, 0x1f6cce0, 0xc000ca8150, 0xc0009d7ec0, 0x4, ...)
	/workspace/source/pkg/plugins/trigger/pull-request.go:51 +0xd35
github.com/jenkins-x/lighthouse/pkg/plugins/trigger.handlePullRequest(0x1f999c0, 0xc00025a0c0, 0xc000dde100, 0x0, 0x0, 0x1f6cce0, 0xc000ca8150, 0x1f7f0a0, 0xc000664510, 0x0, ...)
	/workspace/source/pkg/plugins/trigger/trigger.go:146 +0x2b5
github.com/jenkins-x/lighthouse/pkg/webhook.(*Server).HandlePullRequestEvent.func1(0xc000da28a0, 0xc000527378, 0xc0009fc000, 0xc000aeb860, 0x7, 0x1d1e0b8)
	/workspace/source/pkg/webhook/events.go:196 +0x2c2
created by github.com/jenkins-x/lighthouse/pkg/webhook.(*Server).HandlePullRequestEvent
	/workspace/source/pkg/webhook/events.go:188 +0x5bb
```

--- a/pkg/webhook/webhook.go
+++ b/pkg/webhook/webhook.go
@@ -61,6 +61,7 @@ type Options struct {
 	gitServerURL     string
 	configMapWatcher *watcher.ConfigMapWatcher
 	gitClient        git.Client
+	launcher         launcher.PipelineLauncher
 }
 
 // NewCmdWebhook creates the command
@@ -132,6 +133,18 @@ func (o *Options) Run() error {
 
 	o.gitClient = gitClient
 
+	_, jxClient, _, lhClient, _, err := clients.GetClientsAndNamespace(nil)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to create JX client")
+		logrus.Errorf("%s", err.Error())
+		return err
+	}
+	o.launcher, err = launcher.NewLauncher(jxClient, lhClient, o.namespace)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to create PipelineLauncher client")
+		logrus.Errorf("%s", err.Error())
+		return err
+	}
 	mux := http.NewServeMux()
 	mux.Handle(HealthPath, http.HandlerFunc(o.health))
 	mux.Handle(ReadyPath, http.HandlerFunc(o.ready))
@@ -261,6 +274,7 @@ func (o *Options) handleWebHookRequests(w http.ResponseWriter, r *http.Request) 
 		KubernetesClient:  kubeClient,
 		GitClient:         o.gitClient,
 		LighthouseClient:  lhClient.LighthouseV1alpha1().LighthouseJobs(o.namespace),
+		LauncherClient:    o.launcher,
 	}
 	l, output, err := o.ProcessWebHook(logrus.WithField("Webhook", webhook.Kind()), webhook)
 	if err != nil {
@@ -313,11 +327,6 @@ func (o *Options) ProcessWebHook(l *logrus.Entry, webhook scm.Webhook) (*logrus.
 
 		l.Info("invoking Push handler")
 
-		err := o.updateLauncherClientAndReturnError(l, o.server, pushHook.Repository())
-		if err != nil {
-			return l, "", err
-		}
-
 		o.server.HandlePushEvent(l, pushHook)
 		return l, "processed push hook", nil
 	}
@@ -334,11 +343,6 @@ func (o *Options) ProcessWebHook(l *logrus.Entry, webhook scm.Webhook) (*logrus.
 
 		l.Info("invoking PR handler")
 
-		err := o.updateLauncherClientAndReturnError(l, o.server, prHook.Repository())
-		if err != nil {
-			return l, "", err
-		}
-
 		o.server.HandlePullRequestEvent(l, prHook)
 		return l, "processed PR hook", nil
 	}
@@ -352,11 +356,6 @@ func (o *Options) ProcessWebHook(l *logrus.Entry, webhook scm.Webhook) (*logrus.
 		fields["Sender.Name"] = sender.Name
 
 		l.Info("invoking branch handler")
-
-		err := o.updateLauncherClientAndReturnError(l, o.server, branchHook.Repository())
-		if err != nil {
-			return l, "", err
-		}
 
 		o.server.HandleBranchEvent(l, branchHook)
 		return l, "processed branch hook", nil
@@ -378,10 +377,6 @@ func (o *Options) ProcessWebHook(l *logrus.Entry, webhook scm.Webhook) (*logrus.
 
 		l.Info("invoking Issue Comment handler")
 
-		err := o.updateLauncherClientAndReturnError(l, o.server, issueCommentHook.Repository())
-		if err != nil {
-			return l, "", err
-		}
 		o.server.HandleIssueCommentEvent(l, *issueCommentHook)
 		return l, "processed issue comment hook", nil
 	}
@@ -406,10 +401,6 @@ func (o *Options) ProcessWebHook(l *logrus.Entry, webhook scm.Webhook) (*logrus.
 
 		l.Info("invoking Issue Comment handler")
 
-		err := o.updateLauncherClientAndReturnError(l, o.server, prCommentHook.Repository())
-		if err != nil {
-			return l, "", err
-		}
 		o.server.HandlePullRequestCommentEvent(l, *prCommentHook)
 		return l, "processed PR comment hook", nil
 	}
@@ -551,23 +542,6 @@ func (o *Options) createHookServer() (*Server, error) {
 		//TokenGenerator: secretAgent.GetTokenGenerator(o.webhookSecretFile),
 	}
 	return server, nil
-}
-
-func (o *Options) updateLauncherClientAndReturnError(l *logrus.Entry, server *Server, repository scm.Repository) error {
-	_, jxClient, _, lhClient, _, err := clients.GetClientsAndNamespace(nil)
-	if err != nil {
-		err = errors.Wrapf(err, "failed to create JX client")
-		l.Errorf("%s", err.Error())
-		return err
-	}
-	launcherClient, err := launcher.NewLauncher(jxClient, lhClient, o.namespace)
-	if err != nil {
-		err = errors.Wrapf(err, "failed to create PipelineLauncher client")
-		l.Errorf("%s", err.Error())
-		return err
-	}
-	server.ClientAgent.LauncherClient = launcherClient
-	return nil
 }
 
 func responseHTTPError(w http.ResponseWriter, statusCode int, response string) {

